proto/src/determined/experiment/v1/searcher.proto

```diff
@@ -22,12 +22,64 @@ message CompleteValidateAfterOperation {
   double searcher_metric = 2;
 }
 
+// Constant Hyperparameter
+message ConstantHyperparameter {
+  // value of the constant hyperparameter.
+  double val = 1;
+}
+
+// Hyperparameter.
+message Hyperparameter {
+  // A user can provide one of the hyperparameter types for
+  // the custom searcher.
+  oneof union {
+    // Constant hyperparameter.
+    ConstantHyperparameter constant_hyperparam = 1;
+  }
+}
+
+// Create a trial with given hyperparameters.
+message CreateTrialOperation {
+  // The key refers to which part of the model this hyperparameters are for.
+  // An example in this map would be: {"optimizer": {"learning_rate": 0.01}}
+  map<string, Hyperparameter> hyperparams = 1;
+}
+
+// Close a trial with given id.
+message CloseTrialOperation {
+  // trial_id is the id of the trial to close.
+  string trial_id = 1;
+}
+
+// Get progress of custom searcher method.
+message GetProgressOperation {
+  // A message field can't be empty because bindings won't compile.
+  // Making this a message to keep it consistent with other operations.
+  int32 empty_field = 1;
+}
+
+// Shutdown custom searcher method.
+message ShutdownOperation {
+  // A message field can't be empty because bindings won't compile.
+  // Making this a message to keep it consistent with other operations.
+  int32 empty_field = 2;
+}
+
 // SearcherOperation is an operation issued by the searcher.
 message SearcherOperation {
   // A searcher operation is one of the following operations.
   oneof union {
     // ValidateAfter is issued to tell a trial to train some then validate.
     ValidateAfterOperation validate_after = 1;
+    // CreateTrial is issued to create trial.
+    CreateTrialOperation create_trial = 2;
+    // CloseTrial is issued to close trial.
+    CloseTrialOperation close_trial = 3;
+    // GetProgressOperation is issued to get the progress of the custom searcher
```

Comment:
These are messages that go from the custom searcher to the master, so I feel like this one doesn't belong here. Unless it's supposed to be a _report_ of progress, in which case the message type should contain a progress value.